app/components/chat_component.rb

```diff
@@ -0,0 +1,45 @@
+class ChatComponent < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :user, :conversation, :messages, :send_message_id
+
+  def initialize(user:, conversation:, messages:)
+    @user = user
+    @conversation = conversation
+    @messages = messages
```

Comment:
Hmm, storing the messages like this is going to balloon the state of this component. I would consider making your partial partial impure by loading all of the messages on each render and using the broadcast as a signal to re-render.

If you decide to add pagination, I would keep track of that (the current page) in the state of the component.